aten/src/ATen/native/Attention.cpp

```diff
@@ -0,0 +1,18 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <c10/util/Exception.h>
+
+namespace at {
+namespace native {
+
+
+//::std::tuple<at::Tensor,at::Tensor> wrapper__attn(const at::Tensor & q, const at::Tensor & v, const at::Tensor & k)
+std::tuple <Tensor, Tensor> attn(const Tensor& q, const Tensor& k, const Tensor& v) {
+    Tensor k_t = k.t();
```

Comment:
I think you should add some kind of error checking here to make sure that q, k and v have the shape you expect.

Reply:
Made changes based on this suggestion.

Reply:
I meant to add tests in the forward, not the backward.
But that's ok for this lab.